### PR TITLE
Send message when a topic with subscriptions is selected

### PR DESCRIPTION
**Pull request by salmanalibanani**

### CODE REVIEW

File: PurpleExplorer/Views/MainWindow.xaml.cs [hunk 1]
```diff
@@ -32,11 +32,17 @@ private void TreeView_SelectionChanged(object sender, SelectionChangedEventArgs
...
-                var selectedItem = treeView.SelectedItem as ServiceBusSubscription;
-                mainWindowViewModel.SetSelectedSubscription(selectedItem);
+                var selectedSubscription = treeView.SelectedItem as ServiceBusSubscription;
+                mainWindowViewModel.SetSelectedSubscription(selectedSubscription);
+            }
+
+            if (treeView.SelectedItem is ServiceBusTopic)
+            {
+                var selectedTopic = treeView.SelectedItem as ServiceBusTopic;
+                mainWindowViewModel.SetSelectedTopic(selectedTopic);
```
Comment (mderriey) [inline PurpleExplorer/Views/MainWindow.xaml.cs:45]:
```suggestion
            if (treeView.SelectedItem is ServiceBusTopic selectedTopic)
            {
                mainWindowViewModel.SetSelectedTopic(selectedTopic);
```

I understand you were following the style of the previous block, but using that feature remove quite a bit of code.

Comment (mderriey) [reply PurpleExplorer/Views/MainWindow.xaml.cs:45]: You could even use type-based pattern matching in a `switch` statement if you want, <https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/switch#type-pattern>

```csharp
switch (treeView.SelectedItem)
{
    case ServiceBusSubscription selectedSubscription:
        mainWindowViewModel.SetSelectedSubscription(selectedSubscription);
        break;

    case ServiceBusTopic selectedTopic:
        mainWindowViewModel.SetSelectedTopic(selectedTopic);
        break;
}
```

Comment (salmanalibanani) [reply PurpleExplorer/Views/MainWindow.xaml.cs]: Thanks Mick.  I didn't know about it.  Fixed it as you have suggested.

File: PurpleExplorer/Views/MainWindow.xaml.cs [hunk 2]
```diff
@@ -32,11 +32,17 @@ private void TreeView_SelectionChanged(object sender, SelectionChangedEventArgs
             var mainWindowViewModel = DataContext as MainWindowViewModel;
             var treeView = sender as TreeView;
 
-            mainWindowViewModel.ClearSelectedSubscription();
+            mainWindowViewModel.ClearSelection();
             if (treeView.SelectedItem is ServiceBusSubscription)
             {
-                var selectedItem = treeView.SelectedItem as ServiceBusSubscription;
-                mainWindowViewModel.SetSelectedSubscription(selectedItem);
+                var selectedSubscription = treeView.SelectedItem as ServiceBusSubscription;
```
[the remark this answers is not in the document]
Comment (salmanalibanani) [reply PurpleExplorer/Views/MainWindow.xaml.cs]: That's a valid comment.  I have set the topic in MainWindowViewModel.SetSelectedSubscription.

**Review by armannaj — APPROVED** [review]
LGTM

**Pull request merged**